File: 03-frontend-booking-system/booking-system-web/src/app/features/invitations/dtos/invitation-response-dto.ts
import { SlotResponseDto } from "@features/slots/dtos/slot-response-dto";
import { SlotIncludeMode } from "./slot-include-mode";
import { EventResponseDto } from "@features/events/dtos/event-response-dto";

export class InvitationResponseDto {

    id!:number;
    event!: EventResponseDto;
    userId!:number;
    expiresAt!:string;
    maxUsage?: number | null;
    usedCount!: number;
    accessToken!: string;
    slotIncludeMode!: SlotIncludeMode;
    requiredLogin!: boolean;
    maxUsagePerUser?: number | null;
    createdAt!: string;
    slotList! : SlotResponseDto[];

}

File: 02-backend-booking-system/src/main/java/com/littlek4za/booking_system/features/invitation/dto/InvitationResponseDto.java
package com.littlek4za.booking_system.features.invitation.dto;

import java.time.Instant;
import java.util.List;

import com.littlek4za.booking_system.features.event.dto.EventResponseDto;
import com.littlek4za.booking_system.features.invitation.model.SlotIncludeMode;
import com.littlek4za.booking_system.features.slot.dto.SlotResponseDto;

public record InvitationResponseDto(
    Long id,
    EventResponseDto event,
    Long userId,
    Instant expiresAt,
    Integer maxUsage,
    int usedCount,
    String accessToken,
    SlotIncludeMode slotIncludeMode,
    boolean requiredLogin,
    Integer maxUsagePerIdentity,
    Instant createdAt,
    List<SlotResponseDto> slotList

) {

}
